pkg/setup/flags.go

```diff
@@ -39,6 +40,12 @@ func (f *Flags) Register(flags *pflag.FlagSet) {
 		false,
 		"Do not check CA certficate downloaded from cluster (insecure). Applies to Enterprise DC/OS only.",
 	)
+	flags.BoolVar(
+		&f.noPlugin,
+		"no-plugin",
```

Comment:
There were some disagreement from @klueska on the naming here in our sync meeting, mainly because it's not explicit enough on what it is supposed to do.

IMO we shouldn't make flag names too long as it's hurting usability and doesn't fit well with other flags we have : 
```
      --ca-certs string        Specify the path to a file with trusted CAs to verify requests against.
  -h, --help                   help for setup
      --insecure               Allow requests to bypass TLS certificate verification (insecure).
      --name string            Specify a custom name for the cluster.
      --no-check               Do not check CA certficate downloaded from cluster (insecure). Applies to Enterprise DC/OS only.
      --password string        Specify the password on the command line (insecure).
      --password-env string    Specify an environment variable name that contains the password.
      --password-file string   Specify the path to a file that contains the password (insecure).
      --private-key string     Specify the path to a file that contains the service account private key.
      --provider string        Specify the authentication provider to use for login.
      --username string        Specify the username for login.
```

Here we can see that no flag is longer than 2 words, some flags are not that clear until we look at their description (eg. --insecure or --no-check).

I'm not saying that the current naming is the right way to name flags but at least it seems consistent. If we decide introduce a long flag here we should also move towards this approach on other flags too on the long run (eg. consider deprecating some unclear ones).

cc @leemunroe @AmrAbdelrazik

Reply:
There seems to be 2 schools of thoughts here. Below is a comparison of the same flags on different CLIs:

**curl**
```
--insecure       Allow insecure server connections when using SSL
--cacert <file>  CA certificate to verify peer against
```

**kubectl config set-cluster**
```
--insecure-skip-tls-verify   insecure-skip-tls-verify for the cluster entry in kubeconfig
--certificate-authority      Path to certificate-authority file for the cluster entry in kubeconfig
```

**curl** tends to prefer short flag names with clear help usages, while **kubectl** emphasizes on long and explicit flag names, while their help usages are not that informative.

Given that we only have to **learn once** what a flag is about but then **use it many times**, I think we should go with the **curl** approach. This is also in-line with the ones we already have (`--insecure` and `--ca-cert`).